In [33]:
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("data/datos_criminalidad_2019-2024.csv")
data = pd.DataFrame(df)

### Un grupo de datos por dos columnas

> Filtrar los tipos de delitos por género, usando las columnas ´GENERO´y ´DELITO´.

In [37]:
agrupar_por_delito_y_genero = data.groupby(["Delito", "Genero"])
print(pd.DataFrame(agrupar_por_delito_y_genero))

                                   0  \
0              (ASALTO, DESCONOCIDO)   
1                   (ASALTO, HOMBRE)   
2                    (ASALTO, MUJER)   
3           (HOMICIDIO, DESCONOCIDO)   
4                (HOMICIDIO, HOMBRE)   
5                 (HOMICIDIO, MUJER)   
6               (HURTO, DESCONOCIDO)   
7                    (HURTO, HOMBRE)   
8                     (HURTO, MUJER)   
9                (ROBO, DESCONOCIDO)   
10                    (ROBO, HOMBRE)   
11                     (ROBO, MUJER)   
12   (ROBO DE VEHICULO, DESCONOCIDO)   
13        (ROBO DE VEHICULO, HOMBRE)   
14         (ROBO DE VEHICULO, MUJER)   
15  (TACHA DE VEHICULO, DESCONOCIDO)   
16       (TACHA DE VEHICULO, HOMBRE)   
17        (TACHA DE VEHICULO, MUJER)   

                                                    1  
0          Delito             SubDelito       Fech...  
1          Delito             SubDelito       Fech...  
2          Delito             SubDelito       Fech...  
3             D

### Realizar un sort de un grupo de datos y colocar los primeros 10

Count de cada tipo de delito

In [10]:
# Realizar un sort de un grupo de datos y colocar los primeros 10

resultado_sort = (
    data.groupby(["Delito"])
    .count()
    .sort_values(by=["Victima"], ascending=False)["Victima"]
    .head(10)
)
print(resultado_sort)

Delito
ASALTO               49987
HURTO                40557
HOMICIDIO             3426
ROBO                  1741
TACHA DE VEHICULO      322
ROBO DE VEHICULO        17
Name: Victima, dtype: int64


### Realizar un cálculo de un promedio y explicarlo

In [31]:
df_fechas_formateadas = pd.read_csv(
    "./data/datos_criminalidad_2019-2024.csv", parse_dates=["Fecha"], dayfirst=True
)

# Filtrar por el tipo de delito "ASALTO" y el año 2023
dataFiltered2023_Asalto = df_fechas_formateadas[
    (df_fechas_formateadas["Delito"] == "ASALTO")
    & (df_fechas_formateadas["Fecha"].dt.year == 2023)
]

cantidad_registros_total = len(dataFiltered2023_Asalto)
promedio_anual_2023_asaltos = cantidad_registros_total / 12

# Mostrar los registros filtrados
# print(dataFiltered2023_Asalto)
print(f"Cantidad total de asaltos en 2023: {cantidad_registros_total}")
print(f"Promedio mensual de asaltos en 2023: {promedio_anual_2023_asaltos}")

Cantidad total de asaltos en 2023: 9455
Promedio mensual de asaltos en 2023: 787.9166666666666


Entre los distintos tipos de delitos, se calculó el promedio anual del delito "Asalto", resultando en una cantidad promedio de 787.9 asaltos mensuales. 